src/query/storages/fuse/src/operations/delete.rs

```diff
@@ -297,33 +298,37 @@ impl FuseTable {
         base_segments: Vec<Location>,
         pipeline: &mut Pipeline,
     ) -> Result<()> {
-        match pipeline.pipes.last() {
-            None => Err(ErrorCode::Internal("The pipeline is empty.")),
-            Some(pipe) if pipe.output_size() == 0 => {
-                Err(ErrorCode::Internal("The output of the last pipe is 0."))
-            }
-            Some(pipe) => {
-                let input_size = pipe.output_size();
-                let mut inputs_port = Vec::with_capacity(input_size);
-                for _ in 0..input_size {
+        if pipeline.is_empty() {
+            return Err(ErrorCode::Internal("The pipeline is empty."));
+        }
+
+        match pipeline.output_len() {
+            0 => Err(ErrorCode::Internal("The output of the last pipe is 0.")),
+            last_pipe_size => {
+                let mut inputs_port = Vec::with_capacity(last_pipe_size);
+                for _ in 0..last_pipe_size {
                     inputs_port.push(InputPort::create());
                 }
                 let output_port = OutputPort::create();
-                let processor = MutationTransform::try_create(
-                    ctx,
-                    self.schema(),
-                    inputs_port.clone(),
-                    output_port.clone(),
-                    self.get_operator(),
-                    self.meta_location_generator().clone(),
-                    base_segments,
-                    self.get_block_compact_thresholds(),
-                )?;
-                pipeline.pipes.push(Pipe::ResizePipe {
-                    inputs_port,
-                    outputs_port: vec![output_port],
-                    processor,
+                pipeline.add_pipe(Pipe {
+                    output_length: 1,
+                    input_length: inputs_port.len(),
+                    items: vec![PipeItem {
```

Comment:
It would be better to use a funcation like `PipeItem::new` to create it instead of building the struct.